tools/enso4igv/README.md

```diff
@@ -0,0 +1,133 @@
+citizen of the GraalVM ecosystem, benefits from polyglot capabilities of GraalVM
+as well as its rich tooling offering. One of such tools is _IGV_ - the _Ideal
+Graph Visualizer_ - an excellent tool to get insights into behavior of Graal
+compiler.
+
+This document shows how to use _IGV_ with the _Enso language_.
+
+## Installation
+
+Visit [GraalVM.org](http://graalvm.org) download page and continue towards
+_enterprise edition_ option. There is an _Ideal Graph Visualizer_ option. After
+clicking thru the confirmation dialogs you should get a ZIP - I've just got
+`idealgraphvisualizer-22.1.0.zip` and then:
+
+```bash
+$ unzip idealgraphvisualizer-22.1.0.zip
+$ ./idealgraphvisualizer/bin/idealgraphvisualizer --userdir /tmp/emptyuserdir
+```
+
+launches the _IGV_ application. One doesn't have to use the `--userdir` option,
+but doing so ensures the newly running _IGV_ process is isolated from any
+settings left around by previous usage of _IGV_.
+
+Now download
+[Enso Language Support module](https://github.com/enso-org/enso/actions/workflows/enso4igv.yml).
+Follow the
+[GitHub actions link](https://github.com/enso-org/enso/actions/workflows/enso4igv.yml)
+and select a build (usually the latest one). The build summary page provides
+various information as well as list of artifacts at the bottom. Download the
+_Enso IGV Plugin_ ZIP file. Make sure you are logged into GitHub - artifacts are
+only available to those logged in. Unzip it and get `enso*.nbm` file. This file
+can be installed into _IGV_ (or any other [NetBeans](http://netbeans.apache.org)
+based application). Go to _Tools_/_Plugins_/_Downloaded_ and install the NBM
+file.
+
+![Tools/Plugins/Downloaded](docs/tools_plugins_downloaded.png)
+
+Proceed by clicking _Install_. You may be asked to download _TextMate Lexer_ - a
+necessary dependency of the _Enso support_ module. Continue thru the wizard to
+_finish_ the installation.
+
+![Tools/Plugins/Downloaded](docs/installer.png)
+
+## Using the IGV
+
+Get an instance of the Enso runtime engine (see
+[Running Enso](../../docs/CONTRIBUTING.md#running-enso)) and then launch it with
+special Java options:
+
+```bash
+enso$ JAVA_OPTS="-Dpolyglot.engine.AllowExperimentalOptions=true\
+ -Dpolyglot.engine.TraceCompilation=true\
+ -Dpolyglot.engine.MultiTier=false\
+ -Dgraal.Dump=Truffle:1"\
+ ./built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/bin/enso --run yourprogram.enso
+```
+
+When executed on [GraalVM 21.3.0](http://graalvm.org) these options instruct the
+_Graal/Truffle compiler_ to dump files into `graal_dumps/_sometimestamp_`
+directory. Generating these files takes a while - make sure `yourprogram.enso`
+runs long enough for the system to warmup, compile the code and run at _full
+speed_.
+
+#### Sieve of Eratosthenes Example
+
+As an example you can download
+[sieve.enso](https://github.com/jtulach/sieve/blob/5b32450da35415322e683bb9769aa45f0d71f1df/enso/sieve.enso)
```

Comment:
I agree. It would be good to include [sieve.enso](https://github.com/jtulach/sieve/blob/5b32450da35415322e683bb9769aa45f0d71f1df/enso/sieve.enso) and make it part of daily executed code. That way we know the syntax is correct and the engine can still run the file. Where should I put the `sieve.enso` file then? CCing @hubertp , @jdunkerley , @kustosz - thank you.

Reply:
It cannot be too simple program, it has to warmup. It could be any of the traditional benchmarks like N-body, Funkuch, etc. However I doubt they are simpler.

Anyway let's address this later. I am already behind the schedule and properly solving this (e.g. making sure the file is tested on each change in engine) requires additional thoughts.